### ISSUE TITLE
[brace-style] Add support for `type` declarations

### ISSUE THREAD

**Issue by unao** [typescript-eslint/typescript-eslint#4537]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules":
     {
        // note you must disable the base rule as it can report incorrect errors
       "brace-style": "off",
       "@typescript-eslint/brace-style": [ 2, 'allman' ]
     }
}
```

```TS
type A = { foo: 'bar' }
```

Any type declaration.

**Expected Result**
I would like type declaration to be respected the same way as are other Typescript specific constructs (e.g. `interface`s `enum`s).

For `allman`:
```TS
type A =
{
  foo: 'bar'
}
```

**Actual Result**
Type declarations are ignored by the rule.

**Additional Info**

According to [docs](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/brace-style.md): _It adds support for enum, interface, namespace and module declarations_, so `type`s declarations are not respected by this rule.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^5.10.1` |
| `@typescript-eslint/parser`        | `^5.10.1` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `8.7.0` |
| `node`                             | `v16.13.0` |

**Comment by bradzacher** [comment]
Thanks for the issue!

We won't be adding support for `type` declarations because the base rule doesn't support variables with object literals.
[eslint demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGJyYWNlLXN0eWxlOiBbJ2Vycm9yJywgJ2FsbG1hbiddICovXG5cbmNvbnN0IHggPSB7IGZvbzogJ2JhcicgfTtcbmNvbnN0IHkgPSB7XG4gIGZvbzogJ2Jhcidcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)
